lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -417,13 +482,44 @@ fn generate_constraint_init_group(f: &Field, c: &ConstraintInitGroup) -> proc_ma
                 quote! { #seeds, }
             });
 
+            let validate_pda = {
```

Comment:
Just curious why there's new PDA validation code added here? Was it missing before?

Reply:
Previously, the pda validation here was done in seeds constraint validation function. It made it a bit difficult to decouple the code that way for this PR so I refactored that check into the init group. It's effectively the same.

Reply:
https://github.com/coral-xyz/anchor/pull/2101/files/9daa85cf7d63b6b53e1d9d4be8854628b2a2f214#diff-1773434fcc58d74d77cfb9c10645cc88eeec52f21d891af23b3f783272b7011fR852

Reply:
cool, i'll check it out thanks.

Reply:
I'll need to add it in here as well. https://github.com/coral-xyz/anchor/pull/1938/files#diff-1773434fcc58d74d77cfb9c10645cc88eeec52f21d891af23b3f783272b7011fR394 It's fine just another thing to add to the list of "things to be aware of when writing new macro code" because it's quite important not to omit this check.

Reply:
Ok, I guess this is not so bad. The validation logic here is just needed when using `init` and `seeds` together, to ensure that it's using the canonical seed. It shouldn't be necessary anywhere else, except at init time I guess.